tests/search/booleansearch/booleansearch.rb

```diff
@@ -410,6 +414,25 @@ def assert_search_and_document_summary_for_partial_update_test
                              ["male-SE", "('gender' in ['Male'] or 'country' in ['Sweden'])\n"]])
   end
 
+  def test_predicate_optimizations_are_idempotent
+    set_description("Test that predicate optimizations are idempotent")
+
+    unoptimized_predicate = "not (not (gender in ['Male'] and age in [20..29] and true)) and gender not in ['Female']"
+    doc_id = "unoptimized-1"
+    File.open(@feed_file, "w") {|file| write_doc(file, doc_id, unoptimized_predicate) }
+    deploy_and_feed
+
+    container_port = Environment.instance.vespa_web_service_port
+    optimized_doc = vespa.document_api_v1.get("id:test:test::#{doc_id}", :port => container_port)
+    expected_optimized_predicate = "gender not in [Female] and gender in [Male] and age in [20..29]"
```

Comment:
Missing quotes for the sexes?

Reply:
I also think this could be optimised to `"gender in ['Male'] and age in [20..29]"`, but that's not necessarily what we do :)

Reply:
Quotes are optional and handled by the parser.  I changed the predicate now to have a clause about country instead.